src/main/scala/com/cognite/spark/datasource/NumericDataPointsRdd.scala

```diff
@@ -0,0 +1,123 @@
+package com.cognite.spark.datasource
+
+import cats.implicits._
+import com.cognite.data.api.v2.DataPoints.{NumericDatapoint, TimeseriesData}
+import com.softwaremill.sttp._
+import io.circe.generic.auto._
+import org.apache.spark.{Partition, SparkContext}
+import org.apache.spark.sql.Row
+
+class NumericDataPointsRdd(
+    @transient override val sparkContext: SparkContext,
+    toRow: NumericDatapoint => Row,
+    numPartitions: Int,
+    aggregation: Option[AggregationFilter],
+    granularity: Option[GranularityFilter],
+    minTimestamp: Long,
+    maxTimestamp: Long,
+    getSinglePartitionBaseUri: Uri,
+    config: RelationConfig)
+    extends DataPointsRdd(sparkContext, getSinglePartitionBaseUri, config) {
+  private val granularityMilliseconds = granularityToMilliseconds(granularity)
+  private val unitMilliseconds = granularityToMilliseconds(
+    granularity.map(_.copy(amount = Some(1))))
+  override val timestampLimits: (Long, Long) =
+    (floorToNearest(minTimestamp, unitMilliseconds), ceilToNearest(maxTimestamp, unitMilliseconds))
+
+  override def getDataPointRows(uri: Uri, start: Long): (Seq[Row], Option[Long]) = {
+    val dataPoints = aggregation match {
+      case Some(aggregationFilter) =>
+        val g =
+          granularity.getOrElse(sys.error("Aggregation requested, but no granularity specified"))
+        val uriWithAggregation = uri
+          .param("aggregates", s"${aggregationFilter.aggregation}")
+          .param("granularity", s"${g.amount.getOrElse("")}${g.unit}")
+        getJson[CdpConnector.DataItemsWithCursor[DataPointsItem]](
+          config.apiKey,
+          uriWithAggregation,
+          config.maxRetries)
+          .unsafeRunSync()
+          .data
+          .items
+          .flatMap(dataPoints =>
+            dataPoints.datapoints.map(dataPoint => {
+              NumericDatapoint(
+                dataPoint.timestamp,
+                getAggregationValue(dataPoint, aggregationFilter))
+            }))
+      case None =>
+        getProtobuf[Seq[NumericDatapoint]](config.apiKey, uri, parseResult, config.maxRetries)
+          .unsafeRunSync()
+    }
+    if (dataPoints.lastOption.fold(true)(_.timestamp < start)) {
+      (Seq.empty, None)
+    } else {
+      (dataPoints.map(toRow), dataPoints.lastOption.map(_.timestamp + granularityMilliseconds))
+    }
+  }
+
+  def parseResult(response: Response[Array[Byte]]): Response[Seq[NumericDatapoint]] = {
```

Comment:
The implementation of parseResult is quite similar in NumericDataPointsRdd and StringDataPointsRdd, we could consider implementing it in the superclass.

Reply:
Yes, it's a bit annoying, but I couldn't find a great way to do it: the obvious way would require a type parameter for DataPointsRdd which isn't nice either. I'm going to leave it like this for now.